pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -116,13 +116,20 @@ func (paths *criticalPaths) update(tpVal string, num int32) {
 // (1) critical paths where the least pods are matched on each spread constraint.
 // (2) number of pods matched on each spread constraint.
 type evenPodsSpreadMetadata struct {
+	constraints []topologySpreadConstraint
 	// We record 2 critical paths instead of all critical paths here.
 	// criticalPaths[0].matchNum always holds the minimum matching number.
 	// criticalPaths[1].matchNum is always greater or equal to criticalPaths[0].matchNum, but
 	// it's not guaranteed to be the 2nd minimum match number.
 	tpKeyToCriticalPaths map[string]*criticalPaths
 	// tpPairToMatchNum is keyed with topologyPair, and valued with the number of matching pods.
-	tpPairToMatchNum map[topologyPair]int32
+	tpPairToMatchNum map[topologyPair]int
+}
+
+type topologySpreadConstraint struct {
```

Comment:
One is just the parsed version of the other. I don't see any harm on them having the same name.

Reply:
or, let's put a comment to emphasize that (1) it's an internal version which has a particular `labels.Selector` field instead of `metav1.Selector`, (2) it's for simplifying the matching logic and performance.

Reply:
Done